core/server/master/src/test/java/alluxio/master/file/replication/ReplicationCheckerTest.java

```diff
@@ -418,4 +440,44 @@ public void heartbeatFilesUnderAndOverReplicated() throws Exception {
     Assert.assertEquals(expected2, mMockReplicationHandler.getEvictRequests());
     Assert.assertEquals(expected1, mMockReplicationHandler.getReplicateRequests());
   }
```

Comment:
could we add a testcase where the job keeps failing? after all, this was the failure mode that prompted this change.

Reply:
The job failing case wasn't that interesting. It was really the job service being overloaded with too many jobs which was much more interesting.